app/models/theme.rb

```diff
@@ -70,25 +79,32 @@ def notify_color_change(color)
   end
 
   after_commit ->(theme) do
-    theme.notify_theme_change
-  end, on: :update
+    theme.notify_theme_change(with_scheme: theme.saved_change_to_color_scheme_id?)
+  end, on: [:create, :update]
+
+  def self.get_set_cache(key, &blk)
```

Comment:
Does the key for the cache need to include the current database name? If we're using the same key, the cache might bleed across multisite.